### ISSUE TITLE
Integrate progress from forks

### ISSUE THREAD

**Issue by alerque** [larsks/gitdriver#9]
The are several forks of this project which all fix various issues and extend the features, several of the accomplishing the same thing in different ways or with a different combinations of features.

It would be really nice to get them cleaned up and integrated into one fork. Python 3 support seems to have been done several times, updating rather than starting from scratch on each run has  been done at least twice, there are three iterations of getting author information, mime types have been extended several times, and one [complete repost](https://github.com/deltaidea/gitdrive) (not listed in the forks) has a some Unicode issues worked out and doesn't die on odd entries in the revisions history such as deleted users.

Are you interested in integrating these into this repository? Are you interested in getting help doing that (yes I might volunteer)? Or are you pretty much done with it and should I look into who else wants to collaborate?

**Comment by larsks** [comment]
I am always happy to accept pull requests to any of my projects.  I won't necessarily get to them in a timely manner unless I happen to be actively using the given project.  Having someone other than the submitter confirm that the pr works correctly helps out a bunch.

**Comment by alerque** [comment]
@deltaidea If I rebase some of the things you've integrated in a separate repo onto the original project and submit them as a series of PR's for each addition would you be willing to check them and see that they work? This whole 5 forks with similar features and two base repos thing is kind of a mess and as long as all parties are willing I'd like to see it cleaned up.

**Comment by deltaidea** [comment]
@alerque 
I don't actively use this repo at the moment. Would you like to have write access to my fork so you guys could merge whatever needed?

**Comment by alerque** [comment]
@deltaidea I was actually more interested in getting the work you'd done (along with the other assorted forks out there) merged back into _this_ repository (which is where most people are going to find it). In order to get this done without too many delays from @larsks side another tester would be nice.

By the way part of the problem is that you posted the repository from scratch instead of forking it from here. That makes it hard for new users to discover. I hacked on this one for a long time trying to make it work for my needs before discovering your work.

**Comment by deltaidea** [comment]
Yeah, I have no idea why. Maybe it was late and I just uploaded it again from local clone.
I'll give you guys write access so you can arrange a PR and resolve conflicts. I'm so sorry for the hassle!

**Comment by larsks** [comment]
I will watch here for pull requests.  Thanks for getting involved!
